### PR TITLE
Filter tokens from more logging calls

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Tokens could leak in verbose logs as part of the networking messages (urllib3, among others). This adds the already existing `TokenURLFilters` to more handlers to prevent it.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/common/io.py [hunk 1]
```diff
@@ -323,6 +323,8 @@ def stderr_log_level(level, logger_name=None):
 def attach_stderr_handler(
     level=WARN, logger_name=None, propagate=False, formatter=None
 ):
+    from ..gateways.logging import TokenURLFilter
```
Comment (jezdez) [inline conda/common/io.py:326]: This unfortunately breaks a API promise we've made in https://github.com/conda/conda/blob/7fa76c9665a07687cdea028efa4024a8fc33847d/conda/common/__init__.py#L6-L7

**Review by jezdez — CHANGES_REQUESTED** [review]

File: conda/common/io.py (hunk 1 above)
Comment (jaimergp) [reply conda/common/io.py:326]: 😬 Let me look into it, maybe we can pass it as they are doing with `formatter` too.

Comment (jaimergp) [reply conda/common/io.py]: You can take another look now @jezdez

File: conda/common/io.py [hunk 2]
```diff
@@ -321,7 +321,11 @@ def stderr_log_level(level, logger_name=None):
 
 
 def attach_stderr_handler(
-    level=WARN, logger_name=None, propagate=False, formatter=None
+    level=WARN,
+    logger_name=None,
+    propagate=False,
+    formatter=None,
+    filters=(),
```
[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/common/io.py:328]: I thought the frowned upon thing was mutable values 🤔

File: conda/auxlib/logz.py [hunk 3]
```diff
@@ -172,4 +172,4 @@ def requests_models_Response_builder(builder, response_object):
         builder.append('')  # end with new line
         return "\n".join(builder)
     except Exception as e:
-        log.exception(e)
+        log.exception("%s", e, exc_info=e)
```
[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/auxlib/logz.py:175]: Yes, you are right:

```
>>> from logging import getLogger, basicConfig
>>> basicConfig()
>>> log = getLogger()
>>> try:
...   raise Exception("Message")
... except Exception as e:
...   log.exception(e)
... 
ERROR:root:Message
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception: Message
>>> try:
...   raise Exception("Message")
... except Exception as e:
...   log.exception("%s", e, exc_info=e)
... 
ERROR:root:Message
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception: Message
```

Maybe it saves a couple of searches in the stack but I didn't check. I'll revert.

File: conda/common/io.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/common/io.py]: Anyhow, changed.

**Review by jezdez — APPROVED** [review]

**Pull request merged**